api/capi/include/nntrainer.h

```diff
@@ -215,6 +219,36 @@ int ml_train_model_set_optimizer(ml_train_model_h model,
 int ml_train_model_set_dataset(ml_train_model_h model,
                                ml_train_dataset_h dataset);
 
+/**
+ * @brief Get input dimensions information of the model
+ * @details Use this function to get input dimensions information of the model.
+ * destroy @a dimensions with @a ml_vector_of_tensordims_destroy_func after use.
+ * @a model must be compiled before calling this function.
+ *
+ * @param[in] model The NNTrainer model handle.
+ * @param[out] dimensions The NNtrainer tensor information handle.
+ * @return @c 0 on successs. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid parameter.
+ */
+int ml_train_model_get_input_dimensions(ml_train_model_h model,
```

Comment:
Can you please describe the need for these APIs? I understand that these are useful APIs but as only these 2 are proposed in this PR, I think there might be some exact need for this API which is missing from the description.

Reply:
This would be required if a model user and a model maker is different.

Some scenario would be

```c
ml_train_model_load_from_config('deployed_model.ini');
ml_train_model_get_input_dimensions(model, info); 
user_resize_image_as_per_info(data, info);
```

Reply:
thanks